PWGHF/D2H/Tasks/taskCharmPolarisation.cxx

```diff
@@ -97,68 +112,138 @@ struct TaskPolarisationCharmHadrons {
-    ROOT::Math::PxPyPzMVector fourVecDauCM = boost(fourVecDau);
-    ROOT::Math::XYZVector threeVecDauCM = fourVecDauCM.Vect();
-
-    ROOT::Math::XYZVector randomVec = ROOT::Math::XYZVector(std::sin(thetaRandom) * std::cos(phiRandom), std::sin(thetaRandom) * std::sin(phiRandom), std::cos(thetaRandom));
-    ROOT::Math::XYZVector beamVec = ROOT::Math::XYZVector(0., 0., 1.);
-    ROOT::Math::XYZVector helicityVec = fourVecMother.Vect();
-    ROOT::Math::XYZVector normalVec = ROOT::Math::XYZVector(pyCharmHad, -pxCharmHad, 0.);
-
-    float cosThetaStarHelicity = std::abs(helicityVec.Dot(threeVecDauCM) / std::sqrt(threeVecDauCM.Mag2()) / std::sqrt(helicityVec.Mag2()));
-    float cosThetaStarProduction = std::abs(normalVec.Dot(threeVecDauCM) / std::sqrt(threeVecDauCM.Mag2()) / std::sqrt(normalVec.Mag2()));
-    float cosThetaStarBeam = std::abs(beamVec.Dot(threeVecDauCM) / std::sqrt(threeVecDauCM.Mag2()));
-    float cosThetaStarRandom = std::abs(randomVec.Dot(threeVecDauCM) / std::sqrt(threeVecDauCM.Mag2()));
-
-    if constexpr (withMl) {
-      registry.fill(HIST("hSparseCharmPolarisation"), invMassCharmHadForSparse, candidate.pt(), pzCharmHad, rapidity, cosThetaStarHelicity, cosThetaStarProduction, cosThetaStarBeam, cosThetaStarRandom, outputMl[0], outputMl[1], outputMl[2]);
-    } else {
-      registry.fill(HIST("hSparseCharmPolarisation"), invMassCharmHadForSparse, candidate.pt(), pzCharmHad, rapidity, cosThetaStarHelicity, cosThetaStarProduction, cosThetaStarBeam, cosThetaStarRandom);
-    }
+    } /// end loop over mass hypotheses
   }
 
+  /////////////////////////
+  //   Dstar analysis   ///
+  /////////////////////////
+
   // Dstar with rectangular cuts
   void processDstar(soa::Filtered<CandDstarWSelFlag>::iterator const& dstarCandidate, Tracks const& tracks)
   {
     runPolarisationAnalysis<charm_polarisation::DecayChannel::DstarToDzeroPi, false>(dstarCandidate, tracks);
   }
   PROCESS_SWITCH(TaskPolarisationCharmHadrons, processDstar, "Process Dstar candidates without ML", true);
 
-  // Dstar with rectangular cuts
+  // Dstar with ML cuts
   void processDstarWithMl(soa::Filtered<CandDstarWSelFlag>::iterator const&, Tracks const&)
   {
     // DUMMY
   }
   PROCESS_SWITCH(TaskPolarisationCharmHadrons, processDstarWithMl, "Process Dstar candidates with ML (DUMMY)", false);
+
+  ////////////////////////////
+  //   Lc->pKpi analysis   ///
+  ////////////////////////////
+
+  // Lc->pKpi with rectangular cuts
+  void processLcToPKPi(soa::Filtered<CandLcToPKPiWSelFlag>::iterator const& lcCandidate, Tracks const& tracks)
+  {
+    runPolarisationAnalysis<charm_polarisation::DecayChannel::LcToPKPi, false>(lcCandidate, tracks);
+  }
+  PROCESS_SWITCH(TaskPolarisationCharmHadrons, processLcToPKPi, "Process Lc candidates without ML", false);
+
+  // Lc->pKpi with ML cuts
+  void processLcToPKPiWithMl(soa::Filtered<CandLcToPKPiWSelFlag>::iterator const& lcCandidate, Tracks const& tracks)
```

Comment:
Why didn't you implemented the ML case for Lc->pKpi? It is sufficient to join another table here and call the `runPolarisation` function with the flag `withMl=true`

Reply:
sorry, I got inspired by the D* :D added